keps/sig-windows/0000-20181221-windows-group-managed-service-accounts-for-container-identity.md

```diff
@@ -0,0 +1,225 @@
+* [Alternatives [optional]](#alternatives-optional)
+
+## Summary
+
+Active Directory is a service that is built-in and commonly used on Windows Server deployments for user and computer identity. Apps are run using Active Directory identities to enable common user to service, and service to service authentication and authorization. This proposal aims to support a specific type of identity, Group Managed Service Accounts (GMSA), for use with Windows Server containers. This will allow an operator to choose a GMSA at deployment time, and run containers using it to connect to existing applications such as a database or API server without changing how the authentication and authorization are performed.
+
+## Motivation
+
+There has been a lot of interest in supporting GMSA for Windows containers since it's the only way for a Windows application to use an Active Directory identity. This is shown in asks and questions from both public & private conversations:
+
+- https://github.com/kubernetes/kubernetes/issues/51691 "For windows based containers, there is a need to access shared objects using domain user contexts."
+- Multiple large customers are asking the Microsoft Windows team to enable this feature through container orchestrators
+- Multiple developers have blogged how to use this feature, but all are on standalone machines instead of orchestrated through Kubernetes
+  - https://artisticcheese.wordpress.com/2017/09/09/enabling-integrated-windows-authentication-in-windows-docker-container/
+  - https://community.cireson.com/discussion/3853/example-cireson-scsm-portal-on-docker-windows-containers
+  - https://cloudiqtech.com/windows-2016-docker-containers-using-gmsa-connect-to-sql-server/
+  - https://success.docker.com/api/asset/.%2Fmodernizing-traditional-dot-net-applications%2F%23integratedwindowsauthentication 
+
+
+### Goals
+
+- Windows users can run containers with an existing GMSA identity on Kubernetes
+- No extra files or Windows registry keys are needed on each Windows node. All needed data should flow through Kubernetes+Kubelet APIs
+- Provide a reasonable effort to prevent pods from being inadvertently scheduled with access to a GMSA that the operator shouldn't have rights to
+
+### Non-Goals
+
+- Running Linux applications using GMSA or a general Kerberos based authentication system
+- Replacing any existing Kubernetes authorization or authentication controls
+- Providing unique container identities. By design, Windows GMSA are used where multiple nodes are running apps as the same Active Directory principal.
+- Isolation between container users and processes running as the GMSA. Windows already allows users and system services with sufficient privilege to create processes using a GMSA.
+- Preventing the node from having access to the GMSA. Since the node already has authorization to access the GMSA, it can start processes or services using as the GMSA. Containers do not change this behavior.
```

Comment:
Does this mean that a pod breakout will (in general) compromise all identities in the cluster?

Reply:
A pod breakout will allow a process in the pod to assume any Active Directory GMSA identity that the Windows host machine is authorized to use. Note also that one can also directly directly start containers (say through `docker run ...` outside of Kubernetes) with any GMSA credspec that the machine is authorized to use.